modules/api/src/main/java/org/apache/ignite/sql/async/AsyncResultSet.java

```diff
@@ -50,13 +49,12 @@
  */
 public interface AsyncResultSet {
     /**
-     * Returns metadata for the results if the result contains rows ({@link #hasRowSet()} returns {@code true}), or {@code null} if
-     * inapplicable.
+     * Returns metadata for the results.
      *
      * @return ResultSet metadata.
      * @see ResultSet#metadata()
      */
-    @Nullable ResultSetMetadata metadata();
```

Comment:
I believe, metadata should remain nullable for the public result set, and here is why: in the internal API any execution produces a result represented by a list of lists. This is true for DML and DDL statements as well. In order to be able to consume the result, there is a metadata which describes the type of the rows returned. On the other hand, in public API, results of DDL and DML statements are unwrapped to a particular fields in the ResultSet (affectedRows, wasApplied), hence metadata is only needed to process a result set returned by a query